src/components/side_panel/chart/building_blocks/general_series/general_series_editor.ts

```diff
@@ -0,0 +1,123 @@
+import { Component, useState } from "@odoo/owl";
+import { getNthColor, toHex } from "../../../../../helpers";
+import { getDefinedAxis } from "../../../../../helpers/figures/charts";
+import { _t } from "../../../../../translation";
+import {
+  ChartWithAxisDefinition,
+  DispatchResult,
+  SpreadsheetChildEnv,
+  UID,
+} from "../../../../../types";
+import { SidePanelCollapsible } from "../../../components/collapsible/side_panel_collapsible";
+import { RoundColorPicker } from "../../../components/round_color_picker/round_color_picker";
+import { Section } from "../../../components/section/section";
+
+interface Props {
+  figureId: UID;
+  definition: ChartWithAxisDefinition;
+  updateChart: (figureId: UID, definition: Partial<ChartWithAxisDefinition>) => DispatchResult;
+  canChangeVerticalAxis?: boolean;
+}
+
+export class GeneralSeriesEditor extends Component<Props, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-GeneralSeriesEditor";
+  static components = {
+    SidePanelCollapsible,
+    Section,
+    RoundColorPicker,
+  };
+  static props = {
+    figureId: String,
+    definition: Object,
+    updateChart: Function,
+    canChangeVerticalAxis: { type: Boolean, optional: true },
+    slots: { type: Object, optional: true },
+  };
+
+  private state = useState({ index: -1 });
+
+  getDataSeries() {
+    const runtime = this.env.model.getters.getChartRuntime(this.props.figureId);
+    if (!runtime || !("chartJsConfig" in runtime)) {
+      return [];
+    }
+    return [_t("All series"), ...runtime.chartJsConfig.data.datasets.map((d) => d.label)];
+  }
+
+  updateSerieEditor(ev) {
+    this.state.index = ev.target.selectedIndex - 1;
+  }
+
+  updateDataSeriesColor(color: string) {
+    const dataSets = this.props.definition.dataSets;
+    if (!dataSets?.[this.state.index]) return;
+    dataSets[this.state.index] = {
+      ...dataSets[this.state.index],
+      backgroundColor: color,
+    };
+    this.props.updateChart(this.props.figureId, { dataSets });
+  }
+
+  getDataSerieColor() {
+    if (this.state.index === -1) {
+      return "";
+    }
+    const dataSets = this.props.definition.dataSets;
+    if (!dataSets?.[this.state.index]) return "";
+    const color = dataSets[this.state.index].backgroundColor;
+    return color ? toHex(color) : getNthColor(this.state.index);
+  }
+
+  updateDataSeriesAxis(ev) {
+    const yAxisId = ev.target.value === "left" ? "y" : "y1";
+    const dataSets = this.props.definition.dataSets;
+    const currentIndex = this.state.index;
+    for (let i = 0; i < dataSets.length; i++) {
+      if (currentIndex === -1 || currentIndex === i) {
```

Comment:
My opinion, but reading this loop is a bit confusing IMO. Id go with something like 
```
if(currentIndex === -1) dataset = datasets.map(...) 
else (dataset[currentIndex] = {...}
```